src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -109,19 +109,22 @@ def delete_role_definition(cmd, name, resource_group_name=None, scope=None,
     definitions_client = _auth_client_factory(cmd.cli_ctx, scope).role_definitions
     scope = _build_role_scope(resource_group_name, scope,
                               definitions_client.config.subscription_id)
-    roles = _search_role_definitions(cmd.cli_ctx, definitions_client, name, scope, custom_role_only)
+    roles = _search_role_definitions(cmd.cli_ctx, definitions_client, name, [scope], custom_role_only)
     for r in roles:
         definitions_client.delete(role_definition_id=r.name, scope=scope)
 
 
-def _search_role_definitions(cli_ctx, definitions_client, name, scope, custom_role_only=False):
-    roles = list(definitions_client.list(scope))
-    worker = MultiAPIAdaptor(cli_ctx)
-    if name:
-        roles = [r for r in roles if r.name == name or worker.get_role_property(r, 'role_name') == name]
-    if custom_role_only:
-        roles = [r for r in roles if worker.get_role_property(r, 'role_type') == 'CustomRole']
-    return roles
+def _search_role_definitions(cli_ctx, definitions_client, name, scopes, custom_role_only=False):
+    for scope in scopes:
+        roles = list(definitions_client.list(scope))
+        worker = MultiAPIAdaptor(cli_ctx)
+        if name:
+            roles = [r for r in roles if r.name == name or worker.get_role_property(r, 'role_name') == name]
+        if custom_role_only:
+            roles = [r for r in roles if worker.get_role_property(r, 'role_type') == 'CustomRole']
+        if roles:
+            return roles
```

Comment:
Do we want to return roles and break out of the for loop immediately or search all scopes and keep updating roles?

Reply:
The old behavior is to only update one, which is the main scenario which I would like to maintain.

Reply:
We don't need a history entry since it is a bug fix towards a pretty edge scenario